Creates a default name for the constant based on valid naming characters

LucasTaves opened the pull request:
# Changes
- Uses the string content as a way to name the created constants

Commit:
Creates a default name for the constant based on valid naming characters

arazmakhnin commented on CcWorks/Workers/Solvers/BrpMagicStringsSolver.cs:
```diff
@@ -168,6 +171,18 @@ private static List<string> AnalyzeMagicStrings(SyntaxNode node)
+        {
+            var trimmed = Regex.Replace(
+                constantSubstring.Trim(),
+                @"[^\w]",
```
`\W` can be used here instead of `[^\w]`

arazmakhnin commented on CcWorks/Workers/Solvers/BrpMagicStringsSolver.cs:
```diff
@@ -168,6 +171,18 @@ private static List<string> AnalyzeMagicStrings(SyntaxNode node)
             return result;
         }
 
+        private static string GetConstantName(string constantSubstring)
```
This function should also:
- trim leading numbers
- truncate long names to (for example) 30 chars
- check if result name is empty of whitespace, then return default unique name

arazmakhnin reviewed (changes requested):
Please fix highlighted issues.
Also, there are two additional points:
1. Newly created constants should be differ from existed ones. ICs should know what constants they should check and rename, and what constants they shouldn't touch. Maybe with commented line or something else...
2. Please fix unit tests :)

Commit:
Removes numbers from the string and fallback to old naming when the name can't be inferred from the string

LucasTaves commented:
Didn't saw your comments, fixing it now

Commit:
Fixes and add more tests
Makes the naming remove leading zeroes when determining the name

Commit:
Address review and check for already chosen names to avoid creating duplicate constants

Commit:
Saves the created names, if a second const after being trimmed receives the same name, falls back to default naming.
Looks for all members declared in the class, is the naming derived from the content matches it, fall backs to default naming.

Add tests for the 2 scenarios above

Commit:
Refactored code of GetConstantName function
